modules/admin-ui-frontend/app/scripts/shared/directives/passwordStrengthDirective.js

```diff
@@ -0,0 +1,140 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+/**
+ * @ngdoc directive
+ * @description
+ * Checks the strength of a password.
+ *
+ * @example
+ * <pw-strength ng-model="model.password"></pw-strength>
+ */
+
+'use strict';
+
+angular.module('adminNg.directives')
+    .directive('pwStrength', [function () {
+      return {
+        restrict: 'E',
+        scope: {
+          password: '=ngModel'
+        },
+
+        link: function (scope, elem, attrs, ctrl) {
+          scope.$watch('password', function (pw) {
+
+            scope.rule =
+                requirements(pw && /[A-Z]/.test(pw)) +
+                requirements(pw && /[a-z]/.test(pw)) +
+                requirements(pw && /\d/.test(pw)) +
+                requirements(pw && /(?=.*\W)/.test(pw)) +
+                requirements(pw && /^.{8,}$/.test(pw));
+
+            // bad passwords from https://en.wikipedia.org/wiki/List_of_the_most_common_passwords
+            // plus Opencast's default password
+            const bad_passwords = ['0', '111111', '1111111', '123', '123123', '123321',
+              '1234', '12345', '123456', '1234567', '12345678', '123456789', '1234567890',
+              '12345679', '123qwe', '18atcskd2w', '1q2w3e', '1q2w3e4r', '1q2w3e4r5t',
+              '3rjs1la7qe', '555555', '654321', '666666', '7777777', '888888',
+              '987654321', 'aa12345678', 'abc123', 'admin', 'dragon', 'Dragon', 'google',
+              'iloveyou', 'Iloveyou', 'lovely', 'Monkey', 'mynoob', 'password',
+              'password1', 'password12', 'password123', 'princess', 'qwerty', 'qwerty123', 'qwertyuiop',
+              'Qwertyuiop', 'welcome', 'zxcvbnm', 'opencast' ];
+
+            var bar_color = 'background:white;';
+            var bar_width = 'width:0%;';
+            var bar_text = '';
+            var strength = calcStrength(scope.rule, pw);
+
+            setProgBar(strength);
+
+            function requirements(rule) {
+              return rule ? 1 : 0;
+            }
+
+            function calcStrength(rule, pw){
+
+              if (bad_passwords.indexOf(pw) > -1) {
+                strength = 0;
+                return strength;
+              }
+              // for every unused rule/requirement = -5 points
+              var usedRules = (rule - 5) * 5;
+
+              var uniqueChars = (pw.split('').sort().join('').replace(/(.)\1+/g, '').length) * 2;
+              var pw_length = pw.length * 4;
+              var lowerCase = (pw.length - pw.replace(/[a-z]/g, '').length) * 2;
+              var upperCase = (pw.length - pw.replace(/[A-Z]/g, '').length) * 2;
+              var number = (pw.length - pw.replace(/[0-9]/g, '').length) * 4;
+              var symbol = (pw.length - pw.replace(/\W/g, '').length) * 6;
+
+              var strength = usedRules + uniqueChars + pw_length + lowerCase + upperCase + number + symbol;
+              return strength;
```

Comment:
Let's assume we chose the password `a`.
The password `a`…
- is not in the list of bad passwords. Thus, we continue to the ranking.
- does not match 4 out of the 5 rules, giving us a starting score of `-20`.
- gets `2` points for having one unique character.
- gets `4` points for length.
- gets `2` points for lowercase character length.

We end up with `-20 + 2 + 4 + 2 = -12`.
But `-12` does not match any of your conditions below.
You probably want all of these to be very weak:

```suggestion
              return Math.max(1, usedRules + uniqueChars + pw_length + lowerCase + upperCase + number + symbol);
```